src/main/java/at/ac/tuwien/kr/alpha/common/AtomTranslator.java

```diff
@@ -1,12 +1,68 @@
 package at.ac.tuwien.kr.alpha.common;
 
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+
 import java.util.Iterator;
+import java.util.List;
 
 import static at.ac.tuwien.kr.alpha.common.Literals.atomOf;
 import static at.ac.tuwien.kr.alpha.common.Literals.isNegated;
 
-@FunctionalInterface
+/**
+ * Translates atoms between integer (solver) and object (grounder) representation.
+ */
 public interface AtomTranslator {
+
+	/**
+	 * Returns true whenever the atom is a valid choice point (i.e., it represents a rule body).
+	 * @param atom
+	 * @return
+	 */
+	boolean isAtomChoicePoint(int atom);
+
+	/**
+	 * Returns the highest atomId in use.
+	 * @return the highest atomId in use.
+	 */
+	int getMaxAtomId();
+
+	/**
+	 * Translates an atom represented as int into an Atom object.
+	 * @param atom the atom to translate.
+	 * @return the Atom object represented by the int.
+	 */
+	Atom get(int atom);
+
+	/**
+	 * Translates an atom represented as Atom object into an int.
+	 * @param atom the Atom object to translate.
+	 * @return the int representing the Atom object.
+	 */
+	int get(Atom atom);
+
+	/**
+	 * If the given ground atom is not already stored, associates it with a new integer (ID) and stores it, else
+	 * returns the current associated atom ID. Hence, multiple calls with the same parameter will return the same
+	 * value.
+	 * @param groundAtom the ground atom to look up in the store.
+	 * @return the integer ID of the ground atom, possibly newly assigned.
+	 */
+	int putIfAbsent(Atom groundAtom);
+
+	/**
+	 * Returns whether the given ground atom is known to the AtomTranslator.
+	 * @param groundAtom the ground atom to test.
+	 * @return true if the ground atom is already associated an integer ID.
+	 */
+	boolean contains(Atom groundAtom);
+
+	/**
+	 * Returns a list of currently known but unassigned atoms.
+	 * @param assignment the current assignment.
+	 * @return a list of atoms not having assigned a truth value.
+	 */
+	List<Integer> getUnassignedAtoms(Assignment assignment);
```

Comment:
Indeed, this `getUnassignedAtoms` could be in a better place. Actually, after PR #137 it would be best to place this functionality in the new `TrailAssignment` there.

Note that the `AtomTranslator` is now the interface to the `AtomStore`. Maybe they should be called differently?

Reply:
I agree that `getUnassignedAtoms` should be in `TrailAssignment`.
If `AtomStore` stays the only implementation of `AtomTranslator`, one possibility would be to name the former `AtomStoreImpl` and the latter `AtomStore` ...